hs_core/models.py

```diff
@@ -2342,6 +2342,35 @@ def can_set_file_type(self):
 
     @property
     def url(self):
+        """
+        return the URL of the file contained in this ResourceFile.
+
+        A GET of this URL simply returns the file. This URL is independent of federation.
+        PUT, POST, and DELETE are not supported.
+
+        This choice for a URL is dependent mainly upon conformance to DataOne URL standards
+        that are also conformant to the format in resourcemap.xml. This url does not contain
+        the site URL, which is prefixed when needed.
+
+        This is based upon the resourcemap_urls.py entry:
+
+            url(r'^resource/(?P<shortkey>[0-9a-f-]+)/data/contents/(?.+)/$',
+                views.file_download_url_mapper,
+                name='get_resource_file')
+
+        The important thing is that there is a trailing '/' in that pattern.
+        """
+        # must start with a / in order to concat with current_site_url.
+        return '/' + os.path.join('resource', self.resource.short_id,
+                                  'data', 'contents', self.short_path) + '/'
```

Comment:
@alvacouch Agreed. Let's revert the current change. Updating hs_restclient bit of pain due to PyPI publication. Otherwise, the actual change is one liner.